sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -173,23 +173,28 @@ class GpuSingleDirectoryDataWriter(
       }
       currentWriter.writeAndClose(batch, statsTrackers)
     } else {
-      withResource(batch) { batch =>
+      withResource(batch) { _ =>
+        assertNotExceedMaxFileCounter(
```

Comment:
Can this assertion leverage the `splitIndexes` returned from `getSplitIndexes` ? Because I saw it is to use this split indices in the description of the issue https://github.com/NVIDIA/spark-rapids/issues/6822

Reply:
Good catch!

Reply:
Always generating the `splitIndexes` is fine even it can go beyond the upper limitation. We just make sure to inject the assertion before splitting the table and writing data to files.
Will it compute the file counts twice in the current implementation ? One for the assertion and one for the splitIndexes.

Reply:
Or, we can call this `assertNotExceedMaxFileCounter` inside the `getSplitIndexes`, to reduce the code change for this PR.